File: yoshimoto/plot_chi2.py
import sys
import numpy as np
from scipy.stats import chi2
import matplotlib.pyplot as plt
import my_style

def plot_chi2(ndf=4):

    x_max = 1
    while(True):
        if chi2.sf(x_max, ndf) < 0.00001:
            break
        x_max += 1
    x_max = (x_max // 10) * 10

    x = np.linspace(0, x_max, 500)
    plt.plot(x, chi2.pdf(x, ndf), 'k-')
    plt.title('$\\chi^2$ distribusion ndf = {0}'.format(ndf))
    plt.xlabel("$\\chi^2$")
    plt.ylabel("Probability")
    plt.xlim(0, x_max)
    plt.ylim(0, None)
    plt.show()


if __name__ == "__main__":
    if len(sys.argv) == 1:
        plot_chi2()
    if len(sys.argv) == 2:
        plot_chi2(float(sys.argv[1]))

File: yoshimoto/plot_chi2.cpp
#include "my_style.hpp"
#include <TF1.h>
#include <TMath.h>
#include <sstream>
using namespace std;

void plot_chi2(int ndf = 4)
{
    SetMyStyle();

    int x_max = 1;
    while (true)
    {
        cout << TMath::Prob(x_max, ndf) << endl;
        if (TMath::Prob(x_max, ndf) < 0.00001)
        {
            break;
        }
        x_max++;
    }
    x_max = (x_max / 10) * 10;

    TF1 *pdf = new TF1("pdf", "ROOT::Math::chisquared_pdf(x,[0],0)", 0, x_max);
    pdf->SetParameter(0, ndf);
    pdf->SetNpx(500);
    pdf->SetTitle(Form("#chi^{2} distribusion ndf = %d;#chi^{2};Probability", ndf));
    pdf->Draw();
}